shaman/controllers/repos/flavors.py

```diff
@@ -0,0 +1,47 @@
+from pecan import request, expose
+from shaman.models import Project, Repo
+
+
+class FlavorController(object):
+
+    def __init__(self, flavor_name):
+        self.flavor_name = flavor_name
+        self.project = Project.query.get(request.context['project_id'])
+        self.repo = Repo.query.filter_by(
+            project=self.project,
+            ref=request.context['ref'],
+            sha1=request.context['sha1'],
+            distro=request.context['distro'],
+            distro_version=request.context['distro_version'],
+            flavor=flavor_name).first()
```

Comment:
I'm not sure you can do ``.first()`` here. There is a small chance we might get the exact same repo built on two different chacra nodes.